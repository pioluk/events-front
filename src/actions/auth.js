/* @flow */

import {
  LOGIN_REQUEST,
  LOGIN_SUCCESS,
  LOGIN_FAILURE,
  LOGOUT
} from '../constants/auth'

export function loginRequest (username: string, password: string) {
  return {
    type: LOGIN_REQUEST,
    username,
    password
  }
}

export function loginSuccess (user: string, token: string) {
  return {
    type: LOGIN_SUCCESS,
    user,
    token
  }
}

export function loginFailure (error: Error) {
  return {
    type: LOGIN_FAILURE,
    error
  }
}

export function logout () {
  return {
    type: LOGOUT
  }
}
