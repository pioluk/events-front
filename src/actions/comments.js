// @flow

import {
  FETCH_COMMENTS,
  FETCH_COMMENTS_SUCCESS,
  FETCH_COMMENTS_FAILURE,
  ADD_COMMENT,
  ADD_COMMENT_SUCCESS,
  ADD_COMMENT_FAILURE,
  REMOVE_COMMENT,
  REMOVE_COMMENT_SUCCESS,
  REMOVE_COMMENT_FAILURE,
} from '../constants/comments'

export function fetchComments (eventId: number, limit: number, offset: number) {
  return {
    type: FETCH_COMMENTS,
    eventId,
    limit,
    offset
  }
}

export function fetchCommentsSuccess (count: number, comments: any[]) {
  return {
    type: FETCH_COMMENTS_SUCCESS,
    count,
    comments
  }
}

export function fetchCommentsFailure (error: Error) {
  return {
    type: FETCH_COMMENTS_FAILURE,
    error
  }
}

export function addComment (text: string) {
  return {
    type: ADD_COMMENT,
    text
  }
}

export function addCommentSuccess (comment: any) {
  return {
    type: ADD_COMMENT_SUCCESS,
    comment
  }
}

export function addCommentFailure (error: Error) {
  return {
    type: ADD_COMMENT_FAILURE,
    error
  }
}

export function removeComment (id: number) {
  return {
    type: REMOVE_COMMENT,
    id
  }
}

export function removeCommentSuccess (id: number) {
  return {
    type: REMOVE_COMMENT_SUCCESS,
    id
  }
}

export function removeCommentFailure (error: Error) {
  return {
    type: REMOVE_COMMENT_FAILURE,
    error
  }
}
