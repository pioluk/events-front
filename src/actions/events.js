import {
  FETCH_EVENTS_REQUEST,
  FETCH_EVENTS_SUCCESS,
  FETCH_EVENTS_FAILURE
} from '../constants/events'

export function fetchEventsRequest () {
  return {
    type: FETCH_EVENTS_REQUEST
  }
}

export function fetchEventsSuccess (events) {
  return {
    type: FETCH_EVENTS_SUCCESS,
    events
  }
}

export function fetchEventsFailure (error) {
  return {
    type: FETCH_EVENTS_FAILURE,
    error
  }
}
