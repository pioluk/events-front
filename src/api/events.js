import { apiGet } from './utils'
import { API_URL } from './config'

export const getEvent = id => apiGet('event/' + id)

export const getEvents = () => apiGet('event')

export const createEvent = event => {
  const user = JSON.parse(localStorage.getItem('user')) || {}

  const formData = new FormData()

  const [ startDate, _1 ] = event.dateStart.toJSON().split('T')
  const [ _2, startTime ] = event.timeStart.toJSON().split('T')

  const [ endDate, _3 ] = event.dateEnd.toJSON().split('T')
  const [ _4, endTime ] = event.timeEnd.toJSON().split('T')

  formData.append('UserId', user.id)
  formData.append('title', event.title)
  formData.append('description', event.description)
  formData.append('dateStart', startDate + 'T' + startTime)
  formData.append('dateEnd', endDate + 'T' + endTime)
  formData.append('color', event.color)
  formData.append('image', event.image)
  formData.append('emails', event.emails)
  formData.append('phones', event.phones)
  formData.append('websites', event.websites)
  formData.append('location.name', event.location.name)
  formData.append('location.lat', event.location.lat)
  formData.append('location.lng', event.location.lng)
  formData.append('location.placeId', event.location.placeId)

  return fetch(`${API_URL}/event`, {
    method: 'POST',
    body: formData
  })
  .then(response => {
    if (response.status >= 200 && response.status < 300) {
      return response
    }

    const error = new Error(response.statusText)
    error.response = response
    throw error
  })
  .then(x => x.json())
}
