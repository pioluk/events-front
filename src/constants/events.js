export const FETCH_EVENTS_REQUEST = 'FETCH_EVENTS_REQUEST'
export const FETCH_EVENTS_SUCCESS = 'FETCH_EVENTS_SUCCESS'
export const FETCH_EVENTS_FAILURE = 'FETCH_EVENTS_FAILURE'
export const ADD_EVENT = 'ADD_EVENT'
export const ADD_EVENT_SUCCESS = 'ADD_EVENT_SUCCESS'
export const ADD_EVENT_FAILURE = 'ADD_EVENT_FAILURE'
