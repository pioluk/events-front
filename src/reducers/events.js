/* @flow */

import {
  FETCH_EVENTS_REQUEST,
  FETCH_EVENTS_SUCCESS,
  FETCH_EVENTS_FAILURE,
  FETCH_EVENT_DETAILS_REQUEST,
  FETCH_EVENT_DETAILS_SUCCESS,
  FETCH_EVENT_DETAILS_FAILURE,
  INCREASE_PAGE
} from '../constants/events'

type Event = any

type EventsState = {
  count: number,
  error: ?Error,
  eventComments: Array<any>,
  eventCount: number,
  eventDetails: ?any,
  events: Array<Event>,
  isLoading: boolean,
  page: number
}

const initialState: EventsState = {
  count: 0,
  error: null,
  eventComments: [],
  eventCount: 0,
  eventDetails: null,
  events: [],
  isLoading: false,
  page: 0
}

export default function eventsReducer (state: EventsState = initialState, action: any) {
  switch (action.type) {
    case FETCH_EVENTS_REQUEST:
    case FETCH_EVENT_DETAILS_REQUEST:
      return { ...state, isLoading: true }

    case FETCH_EVENTS_SUCCESS:
      return {
        ...state,
        count: action.count,
        isLoading: false,
        eventCount: state.eventCount + action.events.length,
        events: [...state.events, ...action.events]
      }

    case FETCH_EVENTS_FAILURE:
      return { ...state, isLoading: false, events: [], error: action.error }

    case FETCH_EVENT_DETAILS_SUCCESS:
      return {
        ...state,
        eventDetails: action.eventDetails
      }

    case FETCH_EVENT_DETAILS_FAILURE:
      return {
        ...state,
        error: action.error
      }

    case INCREASE_PAGE:
      return { ...state, page: state.page + 1 }

    default:
      return state
  }
}
