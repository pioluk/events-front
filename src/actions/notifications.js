import { DISPLAY_NOTIFICATION } from '../constants/notifications'

export function displayNotification (message) {
  return {
    type: DISPLAY_NOTIFICATION,
    message
  }
}
