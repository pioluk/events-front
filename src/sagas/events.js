import { call, take, put } from 'redux-saga/effects'
import { FETCH_EVENTS_REQUEST, FETCH_EVENT_DETAILS_REQUEST, ADD_EVENT } from '../constants/events'
import * as actionCreators from '../actions/events'
import { getEvent, getEvents, createEvent } from '../api/events'

export function* getEventsFlow () {
  while (true) {
    yield take(FETCH_EVENTS_REQUEST)
    try {
      const events = yield call(getEvents)
      if (events) {
        yield put(actionCreators.fetchEventsSuccess(events))
      }
    }
    catch (err) {
      yield put(actionCreators.fetchEventsFailure(err))
    }
  }
}

export function* getEventDetailsFlow () {
  while (true) {
    const { eventId } = yield take(FETCH_EVENT_DETAILS_REQUEST)
    try {
      const eventDetails = yield call(getEvent, eventId)
      if (eventDetails) {
        yield put(actionCreators.fetchEventDetailsSuccess(eventDetails))
      }
    }
    catch (err) {
      yield put(actionCreators.fetchEventDetailsFailure(err))
    }
  }
}

export function* createEventFlow () {
  while (true) {
    const { event } = yield take(ADD_EVENT)
    try {
      yield call(createEvent, event)
      yield put(actionCreators.addEventSuccess())
    }
    catch (err) {
      yield put(actionCreators.addEventFailure(err))
    }
  }
}
