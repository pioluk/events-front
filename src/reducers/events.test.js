// global describe,it,expect

import reducer from './events'
import * as actions from '../actions/events'

describe('events reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot()
  })

  it('should handle FETCH_EVENTS_REQUEST', () => {
    const page = 1
    const state = reducer(undefined, actions.fetchEventsRequest(page))
    expect(state).toMatchSnapshot()
  })

  it('should handle FETCH_EVENT_DETAILS_REQUEST', () => {
    const eventId = 1
    const state = reducer(undefined, actions.fetchEventDetailsSuccess(eventId))
    expect(state).toMatchSnapshot()
  })

  it('should handle FETCH_EVENTS_SUCCESS', () => {
    const count = 2
    const events = [
      {
        id: 1,
        UserId: 1,
        title: 'nihil voluptas rerum',
        description: 'Ut ut ipsam. Hic voluptas illo ex voluptatibus dolores aut sed beatae vel. Quo fuga cum animi est voluptatem. Animi numquam magni. Ut labore deleniti tenetur nulla magnam maxime enim. Molestiae omnis harum id sequi accusantium tempora. Tempore est dolorem nesciunt voluptatem quaerat qui dolorum non. Praesentium perferendis ut.',
        dateStart: '2016-05-23T17:38:11.010Z',
        dateEnd: '2016-12-06T02:03:56.063Z',
        color: 'a5dfd8',
        image: null,
        createdAt: '2016-10-26T07:06:39.152Z',
        updatedAt: '2016-10-26T07:06:39.152Z'
      },
      {
        id: 2,
        UserId: 1,
        title: 'voluptatem eos nostrum',
        description: 'Sequi sit minima quae totam dolores suscipit. Vero exercitationem repellendus velit. Earum ullam unde explicabo earum. Corporis architecto ab quisquam consectetur. Odio odit aliquid ex sint consequatur similique. Aut et occaecati saepe rem cupiditate et veritatis aut. Similique molestias eveniet libero labore est laboriosam. Molestiae reiciendis error neque dignissimos.',
        dateStart: '2016-07-04T18:33:59.969Z',
        dateEnd: '2017-08-27T03:37:53.566Z',
        color: '33ed3f',
        image: null,
        createdAt: '2016-10-26T07:06:40.243Z',
        updatedAt: '2016-10-26T07:06:40.243Z'
      }
    ]
    const state = reducer(undefined, actions.fetchEventsSuccess(count, events))
    expect(state).toMatchSnapshot()
  })

  it('should handle FETCH_EVENTS_FAILURE', () => {
    const error = new Error('Fetch events failed.')
    const state = reducer(undefined, actions.fetchEventsFailure(error))
    expect(state).toMatchSnapshot()
  })

  it('should handle FETCH_EVENT_DETAILS_SUCCESS', () => {
    const eventDetails = {
      id: 1,
      UserId: 1,
      title: 'nihil voluptas rerum',
      description: 'Ut ut ipsam. Hic voluptas illo ex voluptatibus dolores aut sed beatae vel. Quo fuga cum animi est voluptatem. Animi numquam magni. Ut labore deleniti tenetur nulla magnam maxime enim. Molestiae omnis harum id sequi accusantium tempora. Tempore est dolorem nesciunt voluptatem quaerat qui dolorum non. Praesentium perferendis ut.',
      dateStart: '2016-05-23T17:38:11.010Z',
      dateEnd: '2016-12-06T02:03:56.063Z',
      color: 'a5dfd8',
      image: null,
      createdAt: '2016-10-26T07:06:39.152Z',
      updatedAt: '2016-10-26T07:06:39.152Z'
    }
    const state = reducer(undefined, actions.fetchEventDetailsSuccess(eventDetails))
    expect(state).toMatchSnapshot()
  })

  it('should handle FETCH_EVENT_DETAILS_FAILURE', () => {
    const error = new Error('Fetch event details failure')
    const state = reducer(undefined, actions.fetchEventDetailsFailure(error))
    expect(state).toMatchSnapshot()
  })

  it('should handle INCREASE_PAGE', () => {
    const state = reducer(undefined, actions.increasePage())
    expect(state).toMatchSnapshot()
  })
})
