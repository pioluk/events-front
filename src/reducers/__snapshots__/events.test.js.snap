exports[`events reducer should handle FETCH_EVENT_DETAILS_FAILURE 1`] = `
Object {
  "count": 0,
  "error": [Error: Fetch event details failure],
  "eventComments": Array [],
  "eventCount": 0,
  "eventDetails": null,
  "events": Array [],
  "isLoading": false,
  "page": 0,
}
`;

exports[`events reducer should handle FETCH_EVENT_DETAILS_REQUEST 1`] = `
Object {
  "count": 0,
  "error": null,
  "eventComments": Array [],
  "eventCount": 0,
  "eventDetails": 1,
  "events": Array [],
  "isLoading": false,
  "page": 0,
}
`;

exports[`events reducer should handle FETCH_EVENT_DETAILS_SUCCESS 1`] = `
Object {
  "count": 0,
  "error": null,
  "eventComments": Array [],
  "eventCount": 0,
  "eventDetails": Object {
    "UserId": 1,
    "color": "a5dfd8",
    "createdAt": "2016-10-26T07:06:39.152Z",
    "dateEnd": "2016-12-06T02:03:56.063Z",
    "dateStart": "2016-05-23T17:38:11.010Z",
    "description": "Ut ut ipsam. Hic voluptas illo ex voluptatibus dolores aut sed beatae vel. Quo fuga cum animi est voluptatem. Animi numquam magni. Ut labore deleniti tenetur nulla magnam maxime enim. Molestiae omnis harum id sequi accusantium tempora. Tempore est dolorem nesciunt voluptatem quaerat qui dolorum non. Praesentium perferendis ut.",
    "id": 1,
    "image": null,
    "title": "nihil voluptas rerum",
    "updatedAt": "2016-10-26T07:06:39.152Z",
  },
  "events": Array [],
  "isLoading": false,
  "page": 0,
}
`;

exports[`events reducer should handle FETCH_EVENTS_FAILURE 1`] = `
Object {
  "count": 0,
  "error": [Error: Fetch events failed.],
  "eventComments": Array [],
  "eventCount": 0,
  "eventDetails": null,
  "events": Array [],
  "isLoading": false,
  "page": 0,
}
`;

exports[`events reducer should handle FETCH_EVENTS_REQUEST 1`] = `
Object {
  "count": 0,
  "error": null,
  "eventComments": Array [],
  "eventCount": 0,
  "eventDetails": null,
  "events": Array [],
  "isLoading": true,
  "page": 0,
}
`;

exports[`events reducer should handle FETCH_EVENTS_SUCCESS 1`] = `
Object {
  "count": 2,
  "error": null,
  "eventComments": Array [],
  "eventCount": 2,
  "eventDetails": null,
  "events": Array [
    Object {
      "UserId": 1,
      "color": "a5dfd8",
      "createdAt": "2016-10-26T07:06:39.152Z",
      "dateEnd": "2016-12-06T02:03:56.063Z",
      "dateStart": "2016-05-23T17:38:11.010Z",
      "description": "Ut ut ipsam. Hic voluptas illo ex voluptatibus dolores aut sed beatae vel. Quo fuga cum animi est voluptatem. Animi numquam magni. Ut labore deleniti tenetur nulla magnam maxime enim. Molestiae omnis harum id sequi accusantium tempora. Tempore est dolorem nesciunt voluptatem quaerat qui dolorum non. Praesentium perferendis ut.",
      "id": 1,
      "image": null,
      "title": "nihil voluptas rerum",
      "updatedAt": "2016-10-26T07:06:39.152Z",
    },
    Object {
      "UserId": 1,
      "color": "33ed3f",
      "createdAt": "2016-10-26T07:06:40.243Z",
      "dateEnd": "2017-08-27T03:37:53.566Z",
      "dateStart": "2016-07-04T18:33:59.969Z",
      "description": "Sequi sit minima quae totam dolores suscipit. Vero exercitationem repellendus velit. Earum ullam unde explicabo earum. Corporis architecto ab quisquam consectetur. Odio odit aliquid ex sint consequatur similique. Aut et occaecati saepe rem cupiditate et veritatis aut. Similique molestias eveniet libero labore est laboriosam. Molestiae reiciendis error neque dignissimos.",
      "id": 2,
      "image": null,
      "title": "voluptatem eos nostrum",
      "updatedAt": "2016-10-26T07:06:40.243Z",
    },
  ],
  "isLoading": false,
  "page": 0,
}
`;

exports[`events reducer should handle INCREASE_PAGE 1`] = `
Object {
  "count": 0,
  "error": null,
  "eventComments": Array [],
  "eventCount": 0,
  "eventDetails": null,
  "events": Array [],
  "isLoading": false,
  "page": 1,
}
`;

exports[`events reducer should return the initial state 1`] = `
Object {
  "count": 0,
  "error": null,
  "eventComments": Array [],
  "eventCount": 0,
  "eventDetails": null,
  "events": Array [],
  "isLoading": false,
  "page": 0,
}
`;
